packages/php-wasm/compile/build-assets/php_wasm.c

```diff
@@ -802,10 +815,40 @@ static void wasm_sapi_register_server_variables(zval *track_vars_array TSRMLS_DC
 	if (value != NULL) {
 		php_register_variable("SCRIPT_NAME", value, track_vars_array TSRMLS_CC);
 		php_register_variable("SCRIPT_FILENAME", value, track_vars_array TSRMLS_CC);
-		php_register_variable("PHP_SELF", value, track_vars_array TSRMLS_CC);
 		php_register_variable("REQUEST_URI", value, track_vars_array TSRMLS_CC);
 	}
 
+	int php_self_set = 0;
+	if (wasm_server_context->document_root != NULL && wasm_server_context->path_translated != NULL)
+	{
+		// Confirm path translated starts with the document root
+		/**
+		 * PHP_SELF is the script path relative to the document rooth.
+		 *
+		 * For example:
+		 * 
+		 * If the document root is /var/www/html and the requested path is /dir/index.php,
+		 * PHP_SELF will be /dir/index.php.
+		 * 
+		 * If the document root is /var/www/html and the requested path /nice/urls is
+		 * served from /var/www/html/another/directory/script.php, PHP_SELF will be
+		 * /another/directory/script.php
+     *
+     * @see https://www.php.net/manual/en/reserved.variables.server.php#:~:text=PHP_SELF
+		 */
+		if (strncmp(wasm_server_context->document_root, wasm_server_context->path_translated, strlen(wasm_server_context->document_root)) == 0) {
+			// Substring of path translated starting after document root
+			char *php_self = wasm_server_context->path_translated + strlen(wasm_server_context->document_root);
+			php_register_variable("PHP_SELF", strdup(php_self), track_vars_array TSRMLS_CC);
```

Comment:
PHP manages memory using the `e` functions, and since we pass this one to PHP, we need to use `estrdup()` here and `efree` later on. I did that locally, the tests passed, I'm rebuilding all PHP versions now.

Reply:
Actually we should still use `free()` to clean up after the regular `strdup()` and not `efree()` . PHP calls `efree()` internally to clean up after the `estrdup()` results we give it.